internal/servers/worker/session.go

```diff
@@ -251,49 +252,132 @@ func (w *Worker) closeConnection(ctx context.Context, req *pbs.CloseConnectionRe
 	return resp, nil
 }
 
-func (w *Worker) closeConnections(ctx context.Context, closeMap map[string]string) error {
-	w.logger.Trace("marking connections as closed", "session_and_connection_ids", fmt.Sprintf("%#v", closeMap))
+// closeConnections is a helper worker function that sends connection
+// close requests to the controller, and sets close times within the
+// worker. It is called during the worker status loop and on
+// connection exit on the proxy.
+//
+// closeInfo is a map of connections mapped to their individual
+// session.
+func (w *Worker) closeConnections(ctx context.Context, closeInfo map[string]string) {
+	if closeInfo == nil {
+		// This should not happen, but it's a no-op if it does. Just
+		// return.
+		return
+	}
+
+	w.logger.Trace("marking connections as closed", "session_and_connection_ids", fmt.Sprintf("%#v", closeInfo))
+	response, err := w.closeConnection(ctx, w.makeCloseConnectionRequest(closeInfo))
```

Comment:
This great, but you could just pull the required map keys via: `reflect.ValueOf(closeInfo).MapKeys()`  I'm not convinced it makes the code more readable though given the need to use the reflect package.  Just a thought... but feel free to ignore my ramblings here.

Reply:
Yeah, I thought about the fact that we use the map without using the values (which is why I noted in the description), and given the narrowness of the scope here hopefully it's not a big deal. Just on a purely pedantic basis I'd be more into hand-rolling a function to collect the keys versus using reflect just to avoid the extra overhead of using the package in general. But that'd not really be relevant here.